File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int currSum = maxSum;
        
        for(int i=1; i < nums.length; i++){
            currSum = Math.max(currSum + nums[i], nums[i]);
            maxSum = Math.max(currSum, maxSum);
        }
        
        return maxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    var max = nums[0]
    var curr = max
    
    for(let i=1; i < nums.length; i++){
        curr = Math.max(curr + nums[i], nums[i])
        max = Math.max(max, curr)
    }
    return max
    
};